### ISSUE TITLE
Client issue when targeting blazor wasm

### ISSUE THREAD

**Issue by ralinator** [CommunityToolkit/Datasync#174]
## Describe the bug

Sorry wasn't sure if this should be a discussion point or an issue as support for blazor wasm isn't included in the supported platforms but I was thinking given that there is a bit of a drive from the MS side with these new templates to have blazor web apps and blazor hybrid apps sharing code etc. that it might be worth considering? Also to note once this one issue was fixed when I was running locally  I was able to get the sync fully working with blazor wasm, there weren't any further blockers in order to get the sync or anything else on the client side working as far as I can tell.

The bug itself manifests when you want to pull/push to the server and it throws an error like below.
![Image](https://github.com/user-attachments/assets/b2139cab-191d-4a4c-90ca-fabbdcf9f139)
If my understanding is correct this is where the exception actually gets thrown
https://source.dot.net/#System.Net.Http/System/Net/Http/[HttpClientHandler.cs](https://source.dot.net/#System.Net.Http/System/Net/Http/HttpClientHandler.cs,141),141
Then on the datasync repo it is trying to do this set in here https://github.com/CommunityToolkit/Datasync/blob/main/src/CommunityToolkit.Datasync.Client/Http/HttpClientExtensions.cs#L37

## To Reproduce

I'm happy to create a minimal repro if required, just wanted to check first if doing the required fix for this project is even desired given it isn't a supported platform

1. Create a solution using a blazor webassembly project as the client with sqlite set up (something similar to this https://github.com/[DavidH102/DotNet8-Blazor-WebAPP-with-SQLite-WASM](https://github.com/DavidH102/DotNet8-Blazor-WebAPP-with-SQLite-WASM), mine was a little different in a private personal project)
2. Set up datasync using the standard documentation
3. Try to pull/push from the server

## Expected behavior

For the code that is looking to set automatic decompression to conditionally ignore this part of the code that throws the platformnotsupportedexception based on the platform. A fix that I have tested locally in the HttpClientFactory is the below - 
![Image](https://github.com/user-attachments/assets/368375e4-6366-4524-89af-e0bbc560dd4d)

## What platforms?

> **Note**: Any bug or feature request that is opened for an unsupported environment will be automatically closed.

* Server:
  * Version of dotnet being used to compile? dotnet 8
  * Library versions? 8.0.4
  * What database are you using? Cosmos DB
  * Where are you running the server? Locally in kestrel
  * GitHub repository containing the code (optional, but helps!) Is in a private repo but happy to create minimal repro

* Client:
  * What platform (Android, iOS, Windows, etc.) versions are you running on? MS Edge on windows
  * Does it happen in an emulator / simulator, or only on a real device? N/A
  * Version of dotnet being used to compile? dotnet 8
  * .NET Runtime Environment (WPF, UWP, WinUI3, MAUI, etc.): Blazor WASM
  * Datsync Toolkit NuGet versions? 8.0.4
  * GitHub repository containing the code (optional, but helps!) Is in a private repo but happy to create minimal repro

**Comment by adrianhall** [comment]
Please create a minimal repro so we can write the appropriate validation logic and unit tests.

**Comment by adrianhall** [comment]
… and, FWIW, I doubt this is the only Blazor WASM issue.  We don’t test on Blazor WASM, so there may be further edge cases.

**Comment by ralinator** [comment]
Think this should give you a minimal repro, only thing you might have to do before running is check the box in VS installer for ".NET 8.0 WebAssembly Build Tools". Other than that should be a case of running it and clicking the button on the home page.
https://github.com/ralinator/DataSyncWasmIssueRepro
![Image](https://github.com/user-attachments/assets/2f1f34d5-866c-403e-8e8c-5dc23b39249b)

**Comment by ralinator** [comment]
Not sure how much help I'll be but thought I would do a little playing around with this, as I somewhat expected doing tests in a bunit type test project didn't seem very helpful since it just seems to run on the operating system of the machine rather than in the browser context (windows for example).
Therefore it seems the only option if we want to try and get these edge cases where running in the browser does have different behaviour is to use some kind of end to end test runner. See example of a quick test using playwright against that minimal repro I did - 
![Image](https://github.com/user-attachments/assets/ed6a8410-9681-4ba2-a7a1-fce5a2ad40e6)
From my limited knowledge of the project, the downsides I can think of implementing this are - 
- Adding a blazor wasm project to the tests as a starter, I guess the time consuming part of it would be thinking about how to cover those edge cases
- Adding an end to end test runner, can be unreliable at times
- Some work involved to integrate both of these into the CI pipeline and possibly some work to have a server backend running as well
- Another couple of projects that need maintaining in the main solution

**Comment by adrianhall** [comment]
Agreed - my plan for this (since I have a number of other edge cases I want to test around Android, etc.) is to have a set of tests for each supported environment in their own test suite.  These end-to-end tests would have a container for the server and a specific test runner for Android, Web, WPF, etc. that I can run.

Unfortunately, the project doesn't have that capability today and I have to learn how to do it.  Fortunately for me, I sit in an organization with years of experience doing this - so hopefully I can get some assistance in setting it up.  However, this is a long pole item.  I'm not going to hold 8.0.5 for it.

**Comment by adrianhall** [comment]
Deferring to 9.0.0 to align with the plan for per-platform testing.

**Comment by adrianhall** [comment]
Closing for now since I fixed the issue.  The testing issue will be opened separately.
